packages/core/src/storage/report.ts

```diff
@@ -90,11 +90,12 @@ function explainStorageOperation(op: StorageOperation<StorageField>, ctx: Storag
 
 function explainTypeChange(ch: TypeChange): string {
   switch (ch.kind) {
+    case 'visibility change':
+      return `Bad upgrade visibility changed ${describeTransition(ch.original, ch.updated)}`;
```

Comment:
This is more in line with the rest of the function:

```suggestion
      return `Bad upgrade ${describeTransition(ch.original, ch.updated)}\nDifferent visibility`;

```

Reply:
I agree 100%. I should have used the `function (<parameter types>) {internal|external} [pure|view|payable] [returns (<return types>)]` EBNF.